### PR TITLE
Support `DROP DATABASE`

### LINKED ISSUE #454: `Database` `ObjectType`
Now that `CREATE DATABASE` is supported, it would be great if `ObjectType::Database` were available for `DROP DATABASE`.

### PR DESCRIPTION
**Pull request by linhr**
This fixes #454.

### CODE REVIEW

File: tests/sqlparser_postgres.rs [hunk 1]
```diff
@@ -893,6 +893,20 @@ fn parse_drop_schema_if_exists() {
     }
 }
 
+#[test]
+fn parse_drop_database_if_exists() {
```
Comment (iffyio) [inline tests/sqlparser_postgres.rs:897]: can this test case also live in common.rs similar to the other drop test case?

Comment (linhr) [reply tests/sqlparser_postgres.rs]: Done!

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Review by alamb — APPROVED** [review]
Thank you @linhr  and @iffyio  🙏

**Comment by alamb** [comment]
🚀

**Pull request merged**